test/adapter/json_api/links_test.rb

```diff
@@ -0,0 +1,37 @@
+require 'test_helper'
+
+LinkPost = Class.new(Model)
+class LinkPostSerializer < ActiveModel::Serializer
+  belongs_to :link_author
+end
+
+LinkAuthor = Class.new(Model)
+class LinkAuthorSerializer < ActiveModel::Serializer
+  links do
+    link :self, href: "//link_author/#{object.id}"
```

Comment:
also, I think the adapter should build the link href for each resource using a uri template and the request object, which it already has